### ISSUE TITLE
[Protocol] Claim & Proof Account Management Interface MVP

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/poktroll#243]
## Objective

Build the foundation for session account management: `mint & burn w.r.t applications & supplies in the context of the claim & proof lifecycle`.

## Origin Document

This is a natural extension of the following tickets:
- #141 
- #142 
- #143 

## Goals

- Decouple the Claim & Proof lifecycle from application/supplier mint & burn flow
- Close the loop on the claim & proof lifecycle
- Design a structure that makes it easy to implement, iterate on and experiment with different types of tokenomics

## Deliverables

- [ ] A single PR that defines a new interface (name TBD) that hooks into post-proof validation account management
- [ ] A single PR that provides the first (basic) implementation of the above interface
- [ ] A single PR that introduces the minimum viable number of on-chain parameters necessary for this implementation
- [ ] Documentation detailing the importance of this interface and how it can be used/extended
- [ ] A PR with at least 1 E2E test validating the results of its behaviour

## Non-goals / Non-deliverables

- #182 

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @Olshansk 
**Co-Owners**: @bryanchriswhite @RawthiL

**Comment by Olshansk** [comment]
As a quick update on this, the plan is to:

1. Scaffold an `accounting` module
2. Have the `accounting` module be dependent on other modules for access to state data
3. The module will primarily be responsible for managing/updating parameters 
4. The module will expose functions (i.e. via `expected_keeper`) that can be called to do post-proof accounting.
5. (4) enables decoupling of accounting and the claim & proof lifecycle, enabling experimentation, mocking and more

Rough Notes (for ref): https://www.notion.so/buildwithgrove/Olshansky-s-Diaries-79fd0008a15049c1b30ea47895649164?pvs=4#9ceaa6f511ab4fb7963f93e794e0736f

cc @RawthiL for visibility

**Comment by Olshansk** [comment]
I tried to run `ignite scaffold module accounting --params compute_to_tokens_multiplier:uint` but the module can't be prefixed with `prefixed with acc`.

@bryanchriswhite I'm going to go with `economics` but going to hold off on the scaffold PR until I get a 👍  in case you have a stronger opinion it.

![Screenshot 2023-12-06 at 7 38 45 PM](https://github.com/pokt-network/poktroll/assets/1892194/64294685-7886-4a59-8307-98a53c9b1c67)
![Screenshot 2023-12-06 at 7 39 01 PM](https://github.com/pokt-network/poktroll/assets/1892194/3eb224b5-75a0-46a6-b104-9f09d6646ce6)
![Screenshot 2023-12-06 at 7 38 54 PM](https://github.com/pokt-network/poktroll/assets/1892194/36068f9e-6e2c-4684-ba3d-f41ec296144a)

**Comment by bryanchriswhite** [comment]
@Olshansk I like economics the most out of those options. What do you think about "treasury"?

![image](https://github.com/pokt-network/poktroll/assets/600733/44f7265d-86e6-4f6f-bc9c-ddea05f1841c)

**Comment by Olshansk** [comment]
tl;dr we're going with `tokenomics`

![giphy](https://github.com/pokt-network/poktroll/assets/1892194/bdbae22a-52d9-48ff-9bed-dd2975c8b289)
